### PR TITLE
Dfr 1497 add ready for hearing state

### PR DESCRIPTION
**Pull request by dawudgovuk**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/DFR-1497

### Change description ###
FR Consent: Create new 'Ready For Hearing' state

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by MindGames2015 — APPROVED** [review]

File: definitions/consented/json/AuthorisationCaseEvent/AuthorisationCaseEvent.json [hunk 1]
```diff
@@ -1447,5 +1447,19 @@
     "CaseEventID": "FR_listForHearing",
     "UserRole": "caseworker-divorce-financialremedy-solicitor",
     "CRUD": "R"
+  },
+  {
+    "LiveFrom": "01/12/2022",
```
Comment (rishikrsharma) [inline definitions/consented/json/AuthorisationCaseEvent/AuthorisationCaseEvent.json:1452]: no new event needed. These event already exists. As permission set for all

**Review by rishikrsharma — CHANGES_REQUESTED** [review]
check my comments

File: definitions/consented/json/CaseEvent/CaseEvent.json [hunk 2]
```diff
@@ -1000,13 +1000,40 @@
...
+    "PostConditionState": "readyForHearing",
     "CallBackURLAboutToSubmitEvent": "${CCD_DEF_COS_URL}/case-orchestration/ccdAboutToSubmitEvent",
     "RetriesTimeoutURLAboutToSubmitEvent": "3,4,5",
     "CallBackURLSubmittedEvent": "${CCD_DEF_COS_URL}/case-orchestration/ccdSubmittedEvent",
     "RetriesTimeoutURLSubmittedEvent": "3,4,5",
     "SecurityClassification": "Public",
     "ShowSummary": "N",
     "ShowEventNotes": "Y"
+  },
+  {
```
Comment (rishikrsharma) [inline definitions/consented/json/CaseEvent/CaseEvent.json:1012]: New event is still here.

File: definitions/consented/json/CaseEvent/CaseEvent.json [hunk 3]
```diff
@@ -821,7 +821,7 @@
     "Name": "Solicitor Upload document",
     "Description": "Solicitor Upload document",
     "DisplayOrder": 48,
-    "PreConditionState(s)": "awaitingInfo",
+    "PreConditionState(s)": "awaitingInfo;readyForHearing",
```
Comment (rishikrsharma) [inline definitions/consented/json/CaseEvent/CaseEvent.json:824]: This seems wrong - as per requirement - this change need on Upload document event

**Review by rishikrsharma — APPROVED** [review]

**Pull request merged**